script.js

```diff
@@ -0,0 +1,146 @@
+// Possible states of the game
+const GameStates = {
+    INPROGRESS: "In Progress",
+    XVICTORY: "X wins",
+    OVICTORY: "O wins",
+    TIE: "Tie"
+}
+let gameState = GameStates.INPROGRESS
+
+const board = document.querySelectorAll('.grid-square')
+
+// Possible states of a square
+const SquareStates = {
+    X: "X",
+    O: "O",
+    EMPTY: "Empty"
+}
+
+const isTaken = (squareId) => {
+    return isX(squareId) || isO(squareId)
+}
+
+const isX = (squareId) => {
+    return !document.getElementById('ximg' + squareId).classList.contains('hidden')
+}
+
+const isO = (squareId) => {
+    return !document.getElementById('oimg' + squareId).classList.contains('hidden')
+}
+
+const getSquareStates = (squareId) => {
+    if (isX(squareId)){
+        return SquareStates.X
+    }
+    else if (isO(squareId)){
+        return SquareStates.O
+    }
+    else {
+        return SquareStates.EMPTY
+    }
+}
+
+const getTurn = () => {
+    turn = 0
+    for (let i = 0; i < board.length; i++){
+        if (isTaken(i)){
+            turn++
+        }
+    }
+    return turn
+}
+
+const updateTurn = () => {
+    if (gameState !== GameStates.INPROGRESS) {
+        return
+    }
+    let turn = getTurn() % 2 === 0 ? "X" : "O"
```

Comment:
it might make sense for the getTurn function to take care of this. Tho it is very cool